Sprint 6 end task.

IvanSimonovDev opened the pull request:
Realized history without repeats where deleting task of any type doesn't break order of remaining tasks . Added tests for this new functionality. Made additional task.

IvanSimonovDev committed:
Changes:

1) Add possibility to remove tasks from history + delete history limit.

2) Add deletion of tasks from history in case when they are deleted from InMemoryTaskManager.

3) Realize DoublyLinkedList for history storing.

4) Change historyStorage to DoublyLinkedList object instead of ArrayList object.

IvanSimonovDev committed:
Changes:

1) Add method equals() to DoublyLinkedList.Node<E>
2) Add tests for updated historyManager and others.

IvanSimonovDev committed:
Changes:

Make additional task.

IvanSimonovDev committed:
Style corrections.

VladimirTitov4 commented on src/InMemoryHistoryManager.java:
```diff
@@ -4,25 +4,33 @@
     public List<Task> getHistory() {
-        return historyStorage;
+        DoublyLinkedList<Short, Task>.Node<Task> currentNode = historyStorage.getHead();
+        ArrayList<Task> result = new ArrayList<>(historyStorage.size());
```
Тип данных для реализации HashMap - выбирайте Map 
То же самое List для ArrayList или LinkedList 
Вот пару ссылок с объявнением

https://softwareengineering.stackexchange.com/a/225677
https://stackoverflow.com/a/20793630/12210331

VladimirTitov4 commented on src/DoublyLinkedList.java:
```diff
@@ -0,0 +1,136 @@
+        return size;
+    }
+
+    public void addLast(K key, V value) {
```
Метод должен принимать только объект задачи, для чего передавать ключ ?

VladimirTitov4 commented on src/DoublyLinkedList.java:
```diff
@@ -0,0 +1,136 @@
+    }
+
+    public void addFirst(K key, V value) {
+        if (innerMap.containsKey(key)) {
```
Не нужно передавать key вы делаете аналог LinkedList, в нем нет никаких ключей

VladimirTitov4 commented on src/DoublyLinkedList.java:
```diff
@@ -0,0 +1,136 @@
+        this.size++;
+    }
+
+    public V remove(K key) {
```
Метод должен принимать объект задачи (или id) хотя здесь id скорее всего и есть ключ, но получается путаница между мапой и листом. Повторюсь, вы делаете аналог LinkedList, и нужно чтобы переменные использовались похожие, удаление по id, добавление объекта в конец или в начало

здесь возможно можно просто переименовать key -> taskId чтобы не было путаницы

IvanSimonovDev committed:
1) Change class DoublyLinkedList
2) Adapt class InMemoryHistoryManager for new version of class DoublyLinkedList

IvanSimonovDev commented:
Замечания исправил.
